### ISSUE TITLE
extra-bids requirements: sessions?

### ISSUE THREAD

**Issue by Remi-Gau** [vistalab/PRFmodel#74]
any reason why the BIDS session entity is REQUIRED in most "APIs"?

this seems like an extra constraint that would force some people extra data wrangling where as there are tools like pybids and or bids-matlab that can be used to query BIDS dataset to help create more flexible APIs.

**Comment by garikoitz** [comment]
Agreed Remi, in the first version we did a minimally BIDS compatible
version.
Right now we are working on a more comprehensive version, not for the
synthetic datasets (which was the original use), but for the experimental
datasets.
We are creating another container called prfprepare. and we are editing
prfanalyze to be compatible.
Can you wait a couple of weeks? If you want to contribute we can have a
small meeting and decide together (you have one opportunity this Wednesday
at 11:00am Central European time, for example, please let me know)
Thanks again for your interest!
Gari

**Comment by Remi-Gau** [comment]
hey @garikoitz 

thanks for the quick response

this can totally wait on my side: this was a more of a general comment on my side as a BIDS maintainer

won't be able to join on Wednesday

but I could join later to give you my 2 cents:
- I think that you can maybe get some inspirations from the BIDS extension (WIP) that tries to define the API of BIDS apps: https://docs.google.com/document/d/104HLZedFtx0TaXEUwd7eyWvJUlc0CcSUtCzwjNgmGxE/edit#heading=h.4k1noo90gelw
- depending on your needs, using bids-matlab could help... or not.

**Comment by Remi-Gau** [comment]
Other things that exist in the BIDS world that you align with:
- `bids-filter-file.json` used by fmriprep to tweak the behavior of what data needs to be ingested
- similar but maybe better for you as this could be incorporated into your `config.json`: use an `Input` key the way the BIDS stats model does it: https://docs.google.com/document/d/1bq5eNDHTb6Nkx3WUiOBgKvLNnaa5OMcGtD0AZ9yms2M/edit#bookmark=id.8zbk1gw8ll6w

**Comment by Remi-Gau** [comment]
oh and of course those are just friendly suggestions: I have 0 idea how far down the road you are to those new features and how much flexibility you have regarding API definition.
